includes/Core/Authentication/Authentication.php

```diff
@@ -362,10 +362,25 @@ function( $routes ) {
 			'googlesitekit_user_data',
 			function( $user ) {
 				if ( $this->profile->has() ) {
-					$profile_data              = $this->profile->get();
-					$user['user']['email']     = $profile_data['email'];
-					$user['user']['picture']   = $profile_data['photo'];
-					$user['user']['full_name'] = $profile_data['full_name'];
+					$profile_data            = $this->profile->get();
+					$user['user']['email']   = $profile_data['email'];
+					$user['user']['picture'] = $profile_data['photo'];
+
+					// Older versions of Site Kit (before 1.86.0) did not
+					// fetch the user's full name, so we need to check for
+					// that attribute before using it.
+					//
+					// If it isn't detected, we schedule the profile to be
+					// updated in the background so it's available ASAP.
+					if ( ! empty( $profile_data['full_name'] ) ) {
+						$user['user']['full_name'] = $profile_data['full_name'];
+					} else {
+						$this->cron_refresh_profile_data(
```

Comment:
This method is only intended to be called in the context of a WP cron event (i.e. it doesn't schedule the background job, but is used by it).

Filters shouldn't really have side effects, so it would be more appropriate to handle this on another action, but that is out of scope here.

If the full name is empty (as it is now and has never been shown), it's not a problem, so displaying it if present is really a progressive enhancement.

Let's simplify this to only safely pass along the name and we can open a new issue to consider potentially refreshing the profile data on a more regular schedule as there could be other reason to refresh it rather than part of it simply being missing 👍

Reply:
Fair enough. I'll adjust the QA Brief and mention this then, because depending on cron frequency it might take awhile to get the full name to show up.